packages/gatsby/src/datastore/in-memory/indexing.ts

```diff
@@ -196,7 +282,14 @@ export const ensureIndexByQuery = (
     getDataStore()
       .iterateNodesByType(nodeTypeNames[0])
       .forEach(node => {
-        addNodeToFilterCache(node, filterPath, filterCache, resolvedNodesCache)
+        addNodeToFilterCache(
+          node,
+          filterPath,
+          filterCache,
+          resolvedNodesCache,
+          indexFields,
+          resolvedFields
+        )
```

Comment:
Now that we're in the file, should we make this an object so order of args doesn't matter

Reply:
Good call!